Add qtum support to trust wallet core library

timemarkovqtum opened the pull request:
Add support in the trust wallet core library for qtum.

Commit:
Add hrp prefix

Commit:
Add hrp test

Commit:
Add qtum coin type

Commit:
Add qtum coin type configuration

Commit:
Add script hash prefix for qtum

Commit:
Update unit tests for qtum

Commit:
Add qtum tests notes

Commit:
Change qtum explorer url

pandazwb commented on src/Coin.cpp:
```diff
@@ -94,6 +94,9 @@ std::string TW::loadAddress(TWCoinType coin, const Data& data) {
     case TWCoinTypeNEO:
         return NEO::Address(data).string();
 
+    case TWCoinTypeQtum:
```
This seems to be removed by upstream. Would you please sync the latest code and resolve conflicts?

timemarkovqtum replied:
Code re-based and updated android and swift tests.

pandazwb reviewed (commented):
Possible missing changes:

swift/Tests/CoinAddressDerivationTests.swift
swift/Tests/SlipTests.swift
android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt
swift/Tests/Blockchains/LitecoinTests.swift
swift/Sources/Addresses/CoinType+Address.swift
tests/CoinTests.cpp

Would you please double check?

Commit:
Add qtum to coin

Commit:
Update swift wrappers

Commit:
Cpp compile fix after rebase

Commit:
Drop P2SH support for Qtum
Drop P2SH support for Qtum TW general coin architecture for P2SH support assume that the prefix is unique per coin, and from the P2SH prefix can also be identified the coin that the address belong. Qtum and Litecoin use the same simbol M for prefix for P2SH address, they both cannot be identified from one prefix. As a result the generated wrappers for mobile applications will not compile.
If architecture changed, deleting this commit will enable the support.

Commit:
Update coin tests

Commit:
Update swift and android tests

Commit:
Add qtum swift tests

Commit:
Add qtum to the list of supported blockchains
